ShareKit

ryang1428 opened the pull request:
First run through implementing URBNShareKit. Supports what is needed for Free People.

ryang1428 committed:
Created Item Provider files & ShareKit header

ryang1428 committed:
Implemented BodyProvider
- Added properties for bodies for different Activity types
- Added pinterest support

ryang1428 committed:
Handling includedActivityTypes

ryang1428 committed:
Added body examples

ryang1428 committed:
Implemented URBNImageProvider

ryang1428 committed:
Implemented UrlProvider

ryang1428 committed:
Refactored example VC name

ryang1428 committed:
Fixes & Changes
- Removed dupe conditional
- Added some info on testing in ExampleVC
- Converted Storyboard to use Size Classes. Removed iPad storyboard
- Renamed body subject

ryang1428 committed:
Added some documentation to the code

ryang1428 commented:
@flippinjoe21 
1. Removed the dupe
2. They are strings 

Also ready to be looked at again. I added some documentation to the code and cleaned up a few things.

ryang1428 committed:
Updated and Podspec

ryang1428 committed:
Modified defaultBody logic & docs

jgrandelli commented on Pod/Classes/URBNBodyProvider.h:
```diff
@@ -0,0 +1,56 @@
+ This text is used for all body properties that are not set, as well as
+    activitity types that are not handled 
+ */
+@property (nonatomic, strong) NSString *defaultBody;
```
We may want to init with a default body so that its a required. Otherwise we are never guaranteed it will be there.

jgrandelli commented on URBNShareKit.podspec:
```diff
@@ -1,28 +1,15 @@
+  s.author           = { "urbn" => "rgarchinsky@urbn.com" }
+  s.source           = { :git => "https://github.com/urbn/URBNShareKit.git", :tag => s.version.to_s }
 
   s.platform     = :ios, '7.0'
```
Did you test this on iOS 7?

ryang1428 replied:
Yes. iOS 7 will not be able to support 3rd party extensions but it will work with the built in extensions

ryang1428 committed:
Refactored init to take defaultBody

ryang1428 committed:
Init changes
- Ensuring the passed defaultBody is not nil
- Made initWithDefaultBody the default initalizer

jgrandelli commented:
@ryang1428 all merged and good to go. Now hook it up with trunk. Add a git tag. Write some documentation in the readme. And make sure you add me as an owner in the pods trunk.

jgrandelli commented:
Well done
